test/e2e/create_test.go

```diff
@@ -58,6 +58,18 @@ func testWindowsNodeCreation(t *testing.T) {
 	// Create a private key secret with the known private key.
 	require.NoError(t, testCtx.createPrivateKeySecret(true), "could not create known private key secret")
 
+	// in case platform=none is detected just wait for Windows worker node to become available
+	if testCtx.CloudProvider.GetType() == config.NonePlatformType {
+		// TODO: setPowerShellDefaultShell for given Windows instance(s)
+		// TODO: validate windows-instances configMap
+		t.Run("[platform=none] Windows instance configured by ConfigMap controller", func(t *testing.T) {
+			err = testCtx.waitForWindowsNodes(gc.numberOfBYOHNodes, false, false, true)
+			assert.NoError(t, err, "Windows node creation failed")
+		})
+		// Expected number of Windows instances configured as worker nodes, return
+		return
+	}
```

Comment:
I'm hesitant about creating another branching path based on another  `if testCtx.CloudProvider.GetType() ==`, this code really belongs as part of the `ConfigMap controller` test.

An alternative way for you to approach this, is to expand the Provider interface to include a new method such as `NewBYOHConfigMap(instanceCount)`. 

For existing providers (AWS,Azure etc.) this method could call a shared function which encapsulates the logic currently within the BYOH tests used to create the VMs and the windows-instances configmap.
https://github.com/openshift/windows-machine-config-operator/blob/bb5f6d48a340569fd203d2de12237df4d445fb57/test/e2e/create_test.go#L129-L149

Then for the new PlatformNone provider you are adding, you can either move the logic you added in the hack script into the go code, or you can assume it was created ahead of time and do a check to ensure the ConfigMap exists, with the correct amount of Windows instances listed

That way within the `ConfigMap controller` test, what can be done is:
```
require.NoError(t,testCtx.CloudProvider.NewBYOHConfigMap(gc.numberOfBYOHNodes))
err = tc.waitForWindowsNodes(gc.numberOfBYOHNodes, false, false, true)
assert.NoError(t, err, "Windows node creation failed")

```
Doing things this way helps smooth out the flow of our tests, and

Reply:
+1

Reply:
@sebsoto Thanks for the feedback. This is good. Will adjust.